django/utils/deconstruct.py

```diff
@@ -23,7 +23,7 @@ def deconstruct(obj):
             and keyword arguments.
             """
             # Fallback version
-            if path:
+            if path and type(obj) is klass:
```

Comment:
Both changes to the `deconstructible` decorator should be tested and moved to a separate commit (preferably even PR).

Reply:
I think a separate PR is smart at this point, yes.